onnxruntime/core/framework/kernel_registry_manager.h

```diff
@@ -90,5 +93,14 @@ class KernelRegistryManager {
   // Each kernel registry may contain kernels from many different providers.
   // in order to search kernels from a specific provider, we have to iterate all its elements
   std::list<std::shared_ptr<KernelRegistry>> custom_kernel_registries_;
+
+  // kernel type str resolver used by kernel registries for kernel matching
+  using KernelTypeStrResolverVariant = std::variant<
```

Comment:
Wondering if this requires a variant given we've only 2 options which can be decided based on the ORT_MINIMAL_BUILD macro?

Reply:
it's either a union of two types in a full build, or just one type in a minimal build. the latter doesn't require a variant but we can make the code more consistent.